### PR TITLE
Fix 'varDsc->lvExactSize == 12' assert.

### LINKED ISSUE #36586: Assertion failed 'varDsc->lvExactSize == 12'
Failure in libraries System.Numerics.Vectors.Tests: release libraries, checked coreclr with:
```
set COMPlus_JitStress=1
```
(note that TieredCompilation is the default 1)

Windows x64

E.g.,
```
build -s libs+libs.tests -runtimeConfiguration checked -c Release -arch x64 /p:ArchiveTests=true
```

(the `/p:ArchiveTests=true` is required to build the RunTests.cmd wrapper scripts)

```
C:\gh\runtime\artifacts\bin\System.Numerics.Vectors.Tests\net5.0-Release>call .\RunTests.cmd -r C:\gh\runtime\artifacts\bin\testhost\net5.0-Windows_NT-Release-x64
```

```
Assert failure(PID 42844 [0x0000a75c], Thread: 13268 [0x33d4]): Assertion failed 'varDsc->lvExactSize == 12' in 'System.Numerics.Tests.Matrix4x4Tests:Matrix4x4CreateFromAxisAngleTest():this' during 'Lowering nodeinfo' (IL size 357)

    File: C:\gh\runtime\src\coreclr\src\jit\compiler.h Line: 3498
    Image: C:\gh\runtime\artifacts\bin\testhost\net5.0-Windows_NT-Release-x64\dotnet.exe
```

@tannergooding @CarolEidt @echesakovMSFT

### PR DESCRIPTION
**Pull request by sandreenko**
It is a small pessimization, but no diffs(benchmarks/libraries),

I had a more ambitious [change ](https://github.com/dotnet/runtime/compare/master...sandreenko:makeLayourAvailableForEachSIMDNode) (based on Tanner's suggestion and ClassLayout added to JitIntrinsic by Carol recently) that required layout for each Intrinsic (SIMD and HW) node to be initialized and it was passing tests. However:

-  I feel like it was too dangerous for the current moment in the release cycle;
-  would require more time to prepare for review;
- the most important, we have an issue with `LCL_FLD` for structs with overlapping fields when we drop `FldSeq` and can't restore `CLASS_HANDLE` after, so it was not clear for me how to keep struct handle in this case.

Fixes #36586.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @briansull @CarolEidt @tannergooding @dotnet/jit-contrib

**Review by CarolEidt — APPROVED** [review]
LGTM - thanks!

File: src/coreclr/src/jit/optcse.cpp [hunk 1]
```diff
@@ -2638,17 +2638,15 @@ class CSE_Heuristic
...
-            assert((structHnd != NO_CLASS_HANDLE) || (cseLclVarTyp != TYP_STRUCT));
-            if (structHnd != NO_CLASS_HANDLE)
-            {
-                m_pCompiler->lvaSetStruct(cseLclVarNum, structHnd, false);
-            }
-#ifdef FEATURE_SIMD
-            else if (varTypeIsSIMD(cseLclVarTyp))
+            if (structHnd == NO_CLASS_HANDLE)
             {
-                m_pCompiler->lvaGetDesc(cseLclVarNum)->lvSIMDType = true;
```
Comment (briansull) [inline src/coreclr/src/jit/optcse.cpp:2649]: Do we still set lvSIMDType to true for the new CSE SIMD LclVars?
Is it done by lvaSetStruct?

File: src/coreclr/src/jit/optcse.cpp [hunk 2]
```diff
@@ -3216,7 +3214,7 @@ bool Compiler::optIsCSEcandidate(GenTree* tree)
 
     // If this is a struct type, we can only consider it for CSE-ing if we can get at
     // its handle, so that we can create a temp.
```
Comment (briansull) [inline src/coreclr/src/jit/optcse.cpp:3216]: // If this is a struct type (or SIMD type),

File: src/coreclr/src/jit/optcse.cpp (hunk 1 above)
Comment (sandreenko) [reply src/coreclr/src/jit/optcse.cpp:2649]: Yes,`lvaSetStruct` handles it:
https://github.com/dotnet/runtime/blob/d69cfa0f1ce8da1c6fc8ff0c2a2e088cb27a7944/src/coreclr/src/jit/lclvars.cpp#L2633-L2637

**Comment by tannergooding** [comment]
👍

**Pull request merged**